### PR TITLE
Various fixes to use JSON serialization

### PR DESCRIPTION
**Pull request by tribal-tec**

### CODE REVIEW

**Comment by tribal-tec** [comment]
FYI @favreau @chevtche

File: CMakeLists.txt [hunk 1]
```diff
@@ -56,6 +58,7 @@ set(FlatBuffers_Sample_Text_SRCS
   include/flatbuffers/flatbuffers.h
   include/flatbuffers/idl.h
   include/flatbuffers/util.h
+  src/base64.cpp
   src/idl_parser.cpp
   src/idl_gen_text.cpp
   samples/sample_text.cpp
```
Comment (eile) [inline CMakeLists.txt]: Link tests against lib and not recompile?

File: src/base64.cpp [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+    char_array_3[0] = (char_array_4[0] << 2) + ((char_array_4[1] & 0x30) >> 4);
+    char_array_3[1] = ((char_array_4[1] & 0xf) << 4) + ((char_array_4[2] & 0x3c) >> 2);
+    char_array_3[2] = ((char_array_4[2] & 0x3) << 6) + char_array_4[3];
+
+    for (j = 0; (j < i - 1); j++) ret += char_array_3[j];
+  }
+
+  return ret;
+}
```
Comment (eile) [inline src/base64.cpp]: This is unlikely to get merged upstream, since google insists on a CLA

**Comment by eile** [comment]
+1

**Pull request merged**